flake.nix

```diff
@@ -0,0 +1,71 @@
+{
+  description = "odri_control_interface";
+
+  inputs = {
+    # TODO: drop `/module` after https://github.com/Gepetto/nix/pull/54
+    gepetto.url = "github:gepetto/nix/module";
+    flake-parts.follows = "gepetto/flake-parts";
+    nixpkgs.follows = "gepetto/nixpkgs";
+    nix-ros-overlay.follows = "gepetto/nix-ros-overlay";
+    systems.follows = "gepetto/systems";
+    treefmt-nix.follows = "gepetto/treefmt-nix";
+    utils.url = "github:Gepetto/nix-lib";
+  };
+
+  outputs =
+    inputs:
+    inputs.flake-parts.lib.mkFlake { inherit inputs; } {
+      systems = import inputs.systems;
+      imports = [ inputs.gepetto.flakeModule ];
+      perSystem =
+        {
+          lib,
+          pkgs,
+          self',
+          ...
+        }:
+        {
+          packages = {
+            default = self'.packages.odri_control_interface;
+            odri_control_interface = pkgs.odri_control_interface.overrideAttrs {
+              version = inputs.utils.lib.rosVersion pkgs ./package.xml;
+              src = lib.fileset.toSource {
+                root = ./.;
+                fileset = lib.fileset.unions [
+                  ./demos
+                  ./include
+                  ./src
+                  ./srcpy
+                  ./CMakeLists.txt
+                  ./package.xml
+                ];
+              };
+            };
+          };
+
+          apps = {
+            testbench = {
+              type = "app";
+              program = "${self'.packages.odri_control_interface}/bin/odri_control_interface_demo_testbench";
+            };
+            solo12 = {
+              type = "app";
+              program = "${self'.packages.odri_control_interface}/bin/odri_control_interface_demo_solo12";
+            };
+          };
+          devShells.default = pkgs.mkShell {
+            buildInputs = [
+              self'.packages.odri_control_interface
+              pkgs.odri_master_board_sdk
+              pkgs.python3Packages.python
+              pkgs.python3Packages.numpy
+            ];
+            shellHook = ''
+              controlSitePackages=`echo ${self'.packages.odri_control_interface}/lib/python*/site-packages | head -n 1`
+              masterboardSitePackages=`echo ${pkgs.odri_master_board_sdk}/lib/python*/site-packages | head -n 1`
+              export PYTHONPATH="$controlSitePackages:$masterboardSitePackages:$PYTHONPATH"
+            '';
```

Comment:
```suggestion
            PYTHONPATH = lib.concatMapStringsSep ":" (x: "${x}/${pkgs.python3.sitePackages}") [
              self'.packages.odri_control_interface
              pkgs.odri_master_board_sdk
            ]
```

Reply:
> Or better: `pkgs.python3.withPackages (_: [ self'.packages.odri_control_interface pkgs.odri_master_board_sdk ])`

so as buildInputs ? i'm not sure about dropping into a python shell directly, because some python scripts require cli args

Reply:
in `packages`, so yes that is kind of buildInputs

Reply:
hmmm unfortunately doing that doesnt seem to add the packages to PYTHONPATH. maybe the packages have to be marked as python packages for this to work? i'm leaving it at `PYTHONPATH =` for now.

Reply:
Ok, yes, to tell nix that a derivation has a python module, we need to either define it with `pkgs.python3Packages.callPackage` instead of `pkgs.callPackage` (so in nixpkgs use `pkgs/top-level/python-packages.nix` instead of `by-name`, or in gepetto/nix put the package in `py-pkgs` instead of `pkgs`), or transform it into a module afterwards with `pkgs.python3Packages.toPythonModule`.

therefore,
here:
```nix
pkgs.python3.withPackages (p: [
  (p.toPythonModule pkgs.odri_master_board_sdk)
  (p.toPythonModule self'.packages.odri_control_interface)
])
```
and/or in gepetto/nix:
```
git mv {,py-}pkgs/odri-masterboard-sdk.nix
git mv {,py-}pkgs/odri-control-interface.nix
```

Reply:
ref. https://nixos.org/manual/nixpkgs/unstable/#topythonmodule-function